From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Fri, 7 Oct 2022 17:48:48 +0000
Subject: [PATCH] libsysprof: Actually set spawnable flags to inherit stdin

The code calculated flags but never actually set them on the spawnable,
so the `inherit-stdin` property did not work.

Fixes: 8799d2f0ca4a6f939e808443db14b266b2142a7f
See: https://gitlab.gnome.org/GNOME/sysprof/-/issues/76#note_1531919
---
 src/libsysprof/sysprof-local-profiler.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libsysprof/sysprof-local-profiler.c b/src/libsysprof/sysprof-local-profiler.c
index 9b77de305360..59b4aba26e63 100644
--- a/src/libsysprof/sysprof-local-profiler.c
+++ b/src/libsysprof/sysprof-local-profiler.c
@@ -666,6 +666,7 @@ sysprof_local_profiler_start_after_auth (SysprofLocalProfiler *self)
 
       g_ptr_array_add (env, NULL);
 
+      sysprof_spawnable_set_flags (spawnable, flags);
       sysprof_spawnable_set_environ (spawnable, (const gchar * const *)env->pdata);
       sysprof_spawnable_append_args (spawnable, (const gchar * const *)priv->spawn_argv);
 
